js/views/modals/purchase/Purchase.js

```diff
@@ -140,19 +148,37 @@ export default class extends BaseModal {
   }
 
   purchaseListing() {
+    // set the address
+    this.order.addAddress(this.shipping.selectedAddress);
     // set the moderator
     this.order.set('moderator', this.moderators.selectedIDs[0]);
+    // set the shipping option
+    const oShipping = this.order.get('items').at(0).get('shipping');
+    const sOption = this.shipping.shippingOptions.selectedOption;
+    oShipping.set({ name: sOption.name, service: sOption.service });
+
+    this.order.set({}, { validate: true });
 
-    $.post({
-      url: app.getServerUrl('ob/purchase'),
-      data: JSON.stringify(this.order.toJSON()),
-    })
-      .done((data) => {
-        console.log(data);
+    if (!this.order.validationError) {
+      $.post({
+        url: app.getServerUrl('ob/purchase'),
+        data: JSON.stringify(this.order.toJSON()),
       })
-      .fail((data) => {
-        console.log(data);
+        .done((data) => {
+          console.log(data);
+        })
+        .fail((data) => {
+          console.log(data);
+        });
+    } else {
+      let errMsg = '';
+      _.each(this.order.validationError, (obj) => {
```

Comment:
This approach is a bit of a deviation from how we display errors elsewhere, which is inline above the respective field that has an error. I suppose if the errors here are quite minimal, we could probably get away with displaying them all in a modal. But on more traditional forms, for consistency (and IMHO, better UX), I would do them inline.

Anyways... have you tested this with some actual errors? I don't think this would work because the error object is not a list, it's an object, for example:

```
{
  fieldName: ['error1', 'error2'...],
  fieldName2: ['error1'],
}
```

You'd probably want something like:

```
Object.keys(this.order.validationError)
  .forEach(errKey => {
    this.order.validationError[errKey].forEach(err => {
      errMsg += `${errKey} - ${err}<br>`;
    });
  });
```